client/state/sites/plans/selectors/is-introductory-offer-applied-to-plan-price.js

```diff
@@ -0,0 +1,16 @@
+import { get } from 'lodash';
+import { getSitePlan } from 'calypso/state/sites/plans/selectors/get-site-plan';
+
+/**
+ * Returns true if the site plan contains introductory offer pricing.
+ *
+ * @param  {object}  state         global state
+ * @param  {number}  siteId        the site id
+ * @param  {string}  productSlug   the plan product slug
+ * @returns {boolean}              True if the site plan contains an introductory offer discount.
+ */
+export function isIntroductoryOfferAppliedToPlanPrice( state, siteId, productSlug ) {
+	const plan = getSitePlan( state, siteId, productSlug );
+
+	return get( plan, 'introductoryOfferRawPrice', -1 ) > 0;
```

Comment:
Minor, but since this is a new file, it would be fairly easy to avoid lodash here (there's a project to slowly remove it from calypso) by using optional chaining.

```suggestion
	return ( plan?. introductoryOfferRawPrice ?? -1 ) > 0;
```